File: javascript/problem042.js
var fs = require('fs');

var data = fs.readFileSync('../textfiles/problem042.txt', 'utf8');
var wordArray = data.split(',');
var array = wordArray.sort();


triangularNumber = [];

function setTriangularNumber(n){
  triangularNumber.push((n*n + n)/2);
}

function getValueFromWord(word){
  var value = 0;
  for(a=1; a<word.length-1; a++){
    value = value + word.charCodeAt(a) - 64;
  }
  return value;
}

for(i=1; i<20; i++){
  setTriangularNumber(i);
}

var result = 0;

for(i=0; i<array.length; i++){
  if(triangularNumber.indexOf(getValueFromWord(array[i])) != -1){
    result = result + 1;
  }
}

console.log(result);

File: python/problem042.py
f = open("../textfiles/problem042.txt", 'r')
data = f.read()
f.close()
data = data.replace("\"", "")
word_array = data.split(',')

triangular_number = []


def set_triangular(n):
    triangular_number.append((n * n + n) / 2)


for i in range(1, 20):
    set_triangular(i)


def get_score(word):
    score = 0
    for a in word:
        score += ord(a) - 64
    return score


result = 0
for i in word_array:
    if triangular_number.count(get_score(i)) == 1:
        result += 1
print(result)
